src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -633,7 +633,23 @@ internal bool NeedsNullableAttribute()
 
         internal abstract bool ApplyNullableTransforms(byte defaultTransformFlag, ImmutableArray<byte> transforms, ref int position, out TypeSymbol result);
 
-        internal abstract TypeSymbol SetUnknownNullabilityForReferenceTypes();
+        internal abstract TypeSymbol SetNullabilityForReferenceTypes(Func<TypeSymbolWithAnnotations, TypeSymbolWithAnnotations> predicate);
+
+        internal TypeSymbol SetUnknownNullabilityForReferenceTypes()
+        {
+            return SetNullabilityForReferenceTypes(setUnknownNullability);
+
+            TypeSymbolWithAnnotations setUnknownNullability(TypeSymbolWithAnnotations type)
```

Comment:
>TypeSymbolWithAnnotations setUnknownNullability(TypeSymbolWithAnnotations type) [](start = 12, length = 79)

Is delegate for the local function going to be cached? If not, consider using a lambda instead.